src/OpenApi/src/OpenApiGenerator.cs

```diff
@@ -359,16 +359,12 @@ private List<OpenApiParameter> GetOpenApiParameters(MethodInfo methodInfo, Endpo
                 throw new InvalidOperationException($"Encountered a parameter of type '{parameter.ParameterType}' without a name. Parameters must have a name.");
             }
 
-            var (isBodyOrFormParameter, parameterLocation) = GetOpenApiParameterLocation(parameter, pattern, disableInferredBody);
+            var (_, parameterLocation) = GetOpenApiParameterLocation(parameter, pattern, disableInferredBody);
 
-            // If the parameter isn't something that would be populated in RequestBody
-            // or doesn't have a valid ParameterLocation, then it must be a service
-            // parameter that we can ignore.
-            if (!isBodyOrFormParameter && parameterLocation is null)
+            if (parameterLocation is null)
```

Comment:
Let's keep a code comment in here that makes it clear we only populate `OpenApiParameters` with a parameter if it has a valid location.